File: backend/src/main/scala/microservices/operations/train/objects/RegisterRailwayManagerPlannerRequest.scala
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class RegisterRailwayManagerPlannerRequest(operatorCode: String, email: String, displayName: String, password: String)
object RegisterRailwayManagerPlannerRequest:
  given sourceEncoder: Encoder[RegisterRailwayManagerPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[RegisterRailwayManagerPlannerRequest] = deriveDecoder

File: frontend/src/microservices/operations/train/objects/RegisterRailwayManagerPlannerRequest.ts
export type RegisterRailwayManagerPlannerRequest = {
  operatorCode: string
  email: string
  displayName: string
  password: string
}

export const registerRailwayManagerPlannerRequestFromJson = (json: string): RegisterRailwayManagerPlannerRequest =>
  JSON.parse(json) as RegisterRailwayManagerPlannerRequest

export const registerRailwayManagerPlannerRequestToJson = (value: RegisterRailwayManagerPlannerRequest): string =>
  JSON.stringify(value)

